code/go/0chain.net/sharder/transaction.go

```diff
@@ -159,45 +157,55 @@ func (sc *Chain) StoreTransactions(b *block.Block) error {
 	return nil
 }
 
-func (sc *Chain) storeTransactions(sTxns []datastore.Entity) error {
+/* storeTransactions - persists given list of transactions and increment TxnsCount of their round */
+func (sc *Chain) storeTransactions(sTxns []datastore.Entity, roundNumber int64) error {
 	txnSummaryMetadata := datastore.GetEntityMetadata("txn_summary")
-	tctx := persistencestore.WithEntityConnection(common.GetRootContext(), txnSummaryMetadata)
-	defer persistencestore.Close(tctx)
-	return txnSummaryMetadata.GetStore().MultiWrite(tctx, txnSummaryMetadata, sTxns)
-}
+	tctx := ememorystore.WithEntityConnection(common.GetRootContext(), txnSummaryMetadata)
+	defer ememorystore.Close(tctx)
 
-var txnSummaryMV = false
-var roundToHashMVTable = "round_to_hash"
+	rtcKey := transaction.BuildSummaryRoundKey(roundNumber)
+	var rtc transaction.RoundTxnsCount
+	err := txnSummaryMetadata.GetStore().Read(tctx, rtcKey, &rtc)
+	if err != nil {
+		rtc.SetKey(rtcKey)
+	}
+	rtc.TxnsCount += len(sTxns)
```

Comment:
We should use `Merge` operation to do `read->modify->write` instead of do `get` and `put`. Check https://github.com/facebook/rocksdb/wiki/Merge-Operator and perhaps do it later after all works well.

Reply:
Yeah sure, merge operation is what should be used here. However, it's not supported by the wrapper we're using. I guess we need to change it if not any other issues will emerge

Reply:
Will double check it and if it's available I'll certainly use it.